src/main/java/com/gregtechceu/gtceu/api/machine/MetaMachine.java

```diff
@@ -487,6 +489,23 @@ protected InteractionResult onScrewdriverClick(Player playerIn, InteractionHand
         return InteractionResult.PASS;
     }
 
+    /***
+     *
+     * @param side  the side attempting to place the cover
+     * @param cover the cover itself trying to be placed
+     * @return Whether the machine allows the cover to be placed on that side
+     */
+    public boolean canAttachCover(Direction side, CoverBehavior cover) {
+        if (side == getFrontFacing()) {
+            if (cover instanceof IIOCover) {
+                return false;
+            } else if (cover instanceof DetectorCover) {
```

Comment:
this else is redundant since the fallback case is true. Idm leaving it if you wanna be extra explicit about it, but something to think about